src/middlewared/middlewared/service.py

```diff
@@ -871,6 +872,32 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, count, timeout):
+        if osc.IS_LINUX:
+            process = run(['ping', '-c', f'{count}', '-w', f'{timeout}', host])
+        else:
+            process = run(['ping', '-c', f'{count}', '-t', f'{timeout}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Str('hostname'),
+        Int('timeout', default=10),  # seconds
+    )
+    def ping_remote(self, hostname, timeout):
+        """
+        Utility method call a ping in the other node.
+        The other node will just return "pong".
+        """
+        ping_host = self._ping_host(hostname, 3, timeout)
```

Comment:
I would remove the `count` argument because we will be waiting no matter what depending on what the timeout is set to.

Also, we should probably cap the `timeout` value to something reasonable?